### PR TITLE
BZ1259531 adding configurable setting for quota syncing

### PR DESCRIPTION
**Pull request by bfallonf**
@thoraxe @derekwaynecarr this PR is in regards to this BZ:

https://bugzilla.redhat.com/show_bug.cgi?id=1259531

I'll comment in the BZ with more info, but can I ask for a tech review? Thanks in advance!

### CODE REVIEW

File: dev_guide/quota.adoc [hunk 1]
```diff
@@ -53,22 +53,23 @@ The following describes the set of limits that may be enforced by a *ResourceQuo
...
 
-Once a quota is created and usage statistics are up-to-date, the project accepts the
-creation of new content. When you create resources, your quota usage is incremented
-immediately upon the request to create or modify the resource. When you delete
-a resource, your quota use is decremented during the next full recalculation of
-quota statistics for the project. As a result, it may take a moment for your quota
-usage statistics to be reduced to their current observed system value when you
-delete resources.
+Once a quota is created and usage statistics are up-to-date, the project accepts
+the creation of new content. When you create resources, your quota usage is
```
Comment (derekwaynecarr) [inline dev_guide/quota.adoc]: Maybe we should say, "When you create or modify resources, your quota..."

File: dev_guide/quota.adoc [hunk 2]
```diff
@@ -130,3 +131,30 @@ resourcequotas          1       1
...
+  - v1beta3
+  - v1
+  apiServerArguments: null
+  controllerArguments:
+    resource-quota-sync-period:
+      - "10s"
+----
+====
+
+Adjusting the regeneration time can be helpful for creating resources and
```
Comment (derekwaynecarr) [inline dev_guide/quota.adoc]: Increasing the sync period places additional load on the openshift master, so any change should be made to balance overall system performance.

**Comment by derekwaynecarr** [comment]
The text is good, but I do think we should caution changing the value too much given that it can increase load on the server.  There is a card to make observation of compute resources being relinquished (pods, cpu, memory) happen more quickly, not yet clear if it will make the next release of kube though.  When that lands, this setting will only have meaning when wanting to observe deletion of less frequently changed resources (replication controllers, services, etc.)

File: dev_guide/quota.adoc (hunk 2 above)
Comment (bfallonf) [inline dev_guide/quota.adoc]: Changed to suggestion.

**Comment by bfallonf** [comment]
@derekwaynecarr Thanks for the notes. I've changed the above sentence, as well as added a note box about having caution when changing the setting. We can look at this again when the card you're talking about is completed and fix anything as necessary.

Can I get an ack that nothing more is needed for this BZ? Thanks.

**Comment by derekwaynecarr** [comment]
Lgtm

File: dev_guide/quota.adoc [hunk 3]
```diff
@@ -106,9 +107,9 @@ resource-quota.json
 
 To apply a quota to a project:
 
-****
+----
 `$ oc create -f resource-quota.json`
```
Comment (adellape) [inline dev_guide/quota.adoc]: Remove backticks.

File: dev_guide/quota.adoc [hunk 4]
```diff
@@ -130,3 +131,37 @@ resourcequotas          1       1
 services                3       5
 ----
 
+[[configuring_quota_sync_period]]
+
+== Configuring the Quota Synchronization Period
+
+When a set of resources are deleted, the synchronization timeframe of resources
+is determined by the `*resource-quota-sync-period*` setting in the
+`*/etc/openshift/master/master-config.yaml*` file. Before your quota usage is
```
Comment (adellape) [inline dev_guide/quota.adoc]: Markup for a file should be:

```
*_/etc/openshift/master/master-config.yaml_*
```

File: dev_guide/quota.adoc [hunk 5]
```diff
@@ -130,3 +131,37 @@ resourcequotas          1       1
...
+restored, you may encounter problems when attempting to reuse the resources.
+Change the `*resource-quota-sync-period:*` setting to have the set of resources
```
Comment (adellape) [inline dev_guide/quota.adoc]: Remove colon.

File: dev_guide/quota.adoc [hunk 6]
```diff
@@ -130,3 +131,37 @@ resourcequotas          1       1
...
+      - "10s"
+----
+====
+
+Adjusting the regeneration time can be helpful for creating resources and
+determining resource usage when automation is used.
+
+[NOTE]
+====
+The `*resource-quota-sync-period:*` setting is designed to balance system
```
Comment (adellape) [inline dev_guide/quota.adoc]: Remove colon.

**Comment by adellape** [comment]
Minor nits, otherwise :+1:.

**Comment by bfallonf** [comment]
@derekwaynecarr Thanks for that. @adellape nits fixed. Unless there's anything else, this is probably ready to merge :bouquet:

**Pull request merged**